### ISSUE TITLE
Lint your code

### ISSUE THREAD

**Issue by ulfryk** [varanthir/to-do#7]
Add one simple build/linting step to your app. You'll need:

- `npm` iniitlised in yor project
- `eslint` npm package installed
- `lint` npm script added to `package.json` file

Good lack @varanthir :)

**Comment by varanthir** [comment]
Looks like I need understand basic git console commands..

**Comment by ulfryk** [comment]
These are not `git` commands ;p

**Comment by varanthir** [comment]
Me n00b.

**Comment by ulfryk** [comment]
http://lmgtfy.com/?q=eslint

**Comment by varanthir** [comment]
Install `eslint` globally or locally in my repo folder?

**Comment by ulfryk** [comment]
First install it globally and use CLI. When code is cleaned we will proceed further.

For now you can just add `.eslintrc.json` so it's clear which rules are used.

**Comment by varanthir** [comment]
Just `npm install -g eslint` and run `eslint script.js`, I got:
`  28:12  error  Parsing error: Unexpected character '`'
[?] 1 problem (1 error, 0 warnings)`
Line 28 in script.js:
`    return `${to2ds(date.getHours())}:${to2ds(date.getMinutes())}`;`
But script works with ho error.

**Comment by ulfryk** [comment]
--> http://eslint.org/docs/user-guide/configuring

> What about ECMAScript 6 support?
> ESLint has full support for ECMAScript 6. By default, this support is off. You can enable ECMAScript 6 support through [configuration](http://eslint.org/docs/user-guide/configuring).

**Comment by ulfryk** [comment]
those things are ECMAScript 6 ( aka ES6 aka ES2015 ):
```javascript
`string templates ${xx}`
() => {}
class Something
const { a, b } = someObject
…etc, etc
```

**Comment by varanthir** [comment]
Ok, so I tried `.eslintrc.json`:
```javascript
{
    "parserOptions": {
        "ecmaVersion": 6
    }
}
```
But now got many errors.

**Comment by ulfryk** [comment]
Paste stack here please :) (no screenshots)

**Comment by ulfryk** [comment]
Use three backticks ( `` ` `` ) as start and end of `monospace` block

**Comment by varanthir** [comment]
```
C:\Users\Janek\Documents\GitHub\to-do>eslint script.js
Cannot read config file: C:\Users\Janek\Documents\GitHub\to-do\.eslintrc.json
Error: Unexpected token ﻿
SyntaxError: Cannot read config file: C:\Users\Janek\Documents\GitHub\to-do\.eslintrc.json
Error: Unexpected token ﻿
    at Object.parse (native)
    at loadJSONConfigFile (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config\config-file.js:121:21)
    at loadConfigFile (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config\config-file.js:213:26)
    at Object.load (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config\config-file.js:494:18)
    at loadConfig (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config.js:64:33)
    at getLocalConfig (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config.js:126:23)
    at Config.getConfig (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\config.js:227:22)
    at processText (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\cli-engine.js:217:27)
    at processFile (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\cli-engine.js:292:18)
    at executeOnFile (C:\Users\Janek\AppData\Roaming\npm\node_modules\eslint\lib\cli-engine.js:676:23)
```

**Comment by ulfryk** [comment]
Looks like problem with file format maybe ? Is it UTF-8 ? Maybe line endings have to be unix not windows ? I don't know ?

Anyway something seems to be wrong with `.eslintrc.json` file

**Comment by varanthir** [comment]
UTF-8

**Comment by ulfryk** [comment]
Paste whole file `.eslintrc.json` content here

**Comment by varanthir** [comment]
```javascript
{
    "parserOptions": {
        "ecmaVersion": 6
    }
}

```

**Comment by ulfryk** [comment]
It seems to be a valid json…

**Comment by varanthir** [comment]
Wrote .eslint.json once again. No errors (no output).

**Comment by ulfryk** [comment]
why have you closed this ?

**Comment by varanthir** [comment]
Oh, sorry.I forgot adding `lint` script to `package.json`.

**Comment by ulfryk** [comment]
Maybe now you can init `package.json` in your project :)

…and add there `eslint` dependency and simple script (`test`) that will call linting.

And you will be able than to add CI step to you project workflow - linting will be automatically checked and PR's will be blocked if linting failed ? Read about Travis CI

**Comment by ulfryk** [comment]
This issue can be closed when #11 is merged…

**Comment by varanthir** [comment]
Fixed
